themes/gatsby-theme-abstract-blog/gatsby-node.js

```diff
@@ -50,35 +93,35 @@ exports.createSchemaCustomization = async ({ actions }) => {
   `)
 }
 
-exports.createPages = async ({ actions, graphql, reporter }, _opts = {}) => {
-  const components = {}
-  const opts = {
-    postPath: _opts.postPath || defaults.postPath,
-    indexPath: _opts.indexPath || defaults.indexPath,
+exports.onCreateWebpackConfig = ({ actions, reporter }, _opts = {}) => {
+  const components = pluginState.components
+
+  if (!components || !components.post || !components.index) {
```

Comment:
Maybe we should process these separately? e.g. why switch `post` to `fallback` if only `index` is missing?